packages/fxa-content-server/app/scripts/views/subscriptions_product_redirect.js

```diff
@@ -15,31 +15,43 @@ class SubscriptionsProductRedirectView extends FormView {
   initialize(options) {
     this._currentPage = options.currentPage;
     this._productId = this._currentPage.split('/').pop();
-    this._queryParams = Url.searchParams(this.window.location.search);
     this.relier.set('subscriptionProductId', this._productId);
 
     this._subscriptionsConfig = {};
     if (options && options.config && options.config.subscriptions) {
       this._subscriptionsConfig = options.config.subscriptions;
     }
 
-    this.mustAuth =
-      !!this._queryParams.signin ||
```

Comment:
I removed the `signin` queryParam here and in the payments server (added largely in PR #9781 ), since IIUC that was introduced under the assumption that we would have the one "product" route on payments, while we later decided to have two payments routes, "product" for authned users and "checkout" for unauthned users.  @chenba , would you agree?

Reply:
It was added to force a sign in.  I think we still need that.  We are going to have a sign in link on the new flow, if we don't have a way of forcing sign in, the user would be taken back to "checkout" since they are not signed in.

Reply:
Hmm yes looking at the current logic, you are right. I'll add it back in... Thank you.

Reply:
I think that will also fix the functional tests.